### PR TITLE
DT-6607 Prevent finished leg from reappearing as current

### PR DESCRIPTION
**Pull request by partisaani**
## Proposed Changes
Added fuzzy logic to legs of interest deduction in order to prevent a previously completed leg from reappearing as current leg.

## Pull Request Check List

  - [ ] A reasonable set of unit tests is included
  - [x] Console does not show new warnings/errors
  - [x] Changes are documented or they are self explanatory
  - [x] This pull request does not have any merge conflicts
  - [x] All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/util/legUtils.js [hunk 1]
```diff
@@ -3,6 +3,26 @@ import get from 'lodash/get';
 import { BIKEAVL_UNKNOWN } from './vehicleRentalUtils';
 import { getRouteMode } from './modeUtils';
 
+function getNestedValue(obj, path) {
```
Comment (Antiik91) [inline app/util/legUtils.js]: Since this is in legUtils, I would recommend to add a small documentation what these functions are and how they can be used.

Comment (partisaani) [reply app/util/legUtils.js]: getNestedValue could also reside in a more common util file if such exists.

File: app/component/itinerary/navigator/hooks/useRealtimeLegs.js [hunk 2]
```diff
@@ -97,26 +101,52 @@ function getLegsOfInterest(legs, time) {
...
+  const nextLegIdx = legs.findIndex(({ start }) => legTime(start) > time);
+  let currentLeg = legs.find(
     ({ start, end }) => legTime(start) <= time && legTime(end) >= time,
   );
+  let previousLeg = legs.findLast(({ end }) => legTime(end) < time);
+  let nextLeg = legs[nextLegIdx];
+
+  // Indices are shifted by one if a previously completed leg reappears as current.
+  if (
+    nextLeg &&
```
Comment (vesameskanen) [inline app/component/itinerary/navigator/hooks/useRealtimeLegs.js]: I guess flapping should be prevented even at the end of the last leg. In that case just return nextLeg = undefined.

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/navigator/hooks/useRealtimeLegs.js (hunk 2 above)
Comment (partisaani) [reply app/component/itinerary/navigator/hooks/useRealtimeLegs.js]: It definitely should.

Comment (partisaani) [reply app/component/itinerary/navigator/hooks/useRealtimeLegs.js]: Fixed in https://github.com/HSLdevcom/digitransit-ui/pull/5201/commits/0d1563ddf8cefece528f4dacf4e2780e2579beb2

**Comment by Antiik91** [comment]
LGTM

**Review by vesameskanen — APPROVED** [review]

**Pull request merged**